### PR TITLE
feat: add PEFT techniques benchmark for conversational AI (#2310)

### LINKED ISSUE #2310: Comparison of Different Fine-Tuning Techniques for Conversational AI
### Feature request

It would be incredibly helpful to have a clear comparison or support for various fine-tuning techniques specifically for conversational AI. This feature could include insights into their strengths, limitations, and ideal use cases, helping practitioners choose the right approach for their needs.

Here’s a list of techniques to consider:

LoRa
AdaLoRa
BONE
VeRa
XLora
LN Tuning
VbLora
HRA (Hyperparameter Regularization Adapter)
IA3 (Input-Aware Adapter)
Llama Adapter
CPT (Conditional Prompt Tuning)etc

### Motivation

With the growing number of fine-tuning techniques for conversational AI, it can be challenging to identify the most suitable approach for specific use cases. A comprehensive comparison of these techniques—highlighting their strengths, limitations, and ideal scenarios—would save time, reduce trial-and-error, and empower users to make informed decisions. This feature would bridge the gap between research and practical application, enabling more effective model customization and deployment.

### Your contribution

I’d be happy to collaborate on this! While I might not have a complete solution right now, I’m willing to contribute by gathering resources, reviewing papers, or helping organize comparisons. If others are interested in teaming up, we could work together on a PR to make this feature happen. Let’s connect and brainstorm how we can tackle this effectively!

### PR DESCRIPTION
**Pull request by sohammmmm10**
Closes #2310

## Summary
Adds a benchmark script that systematically compares 5 PEFT fine-tuning
techniques on conversational AI data, producing a reproducible comparison table.

## Techniques Compared
| Technique | Config Highlights |
|-----------|-------------------|
| LoRA | r=16, alpha=32, q_proj+v_proj |
| AdaLoRA | r=16, alpha=32, adaptive rank |
| IA3 | k_proj, v_proj, down_proj (learned rescaling) |
| VeRA | r=256, shared random matrices |
| LoHa | r=16, alpha=32, Hadamard product |

## Metrics Collected
- Trainable parameters (count and %)
- Peak GPU memory (MB)
- Training time (seconds)
- Train loss / Eval loss
- Throughput (samples/second)

## Usage
```bash
# Run all techniques on Llama-3.2-1B
python examples/benchmarks/benchmark_peft_techniques.py

# Run specific techniques
python examples/benchmarks/benchmark_peft_techniques.py \
    --model_name microsoft/phi-2 \
    --techniques lora adalora vera

# Quick test
python examples/benchmarks/benchmark_peft_techniques.py \
    --techniques lora ia3 --max_samples 500

### CODE REVIEW

**Comment by Adya-Prasad** [comment]
@sohammmmm10 Thanks for your step, but lets make few things clear.
1. First, I hope you have read and understood the resources mention above:  https://huggingface.co/docs/peft/index , https://github.com/huggingface/peft/tree/main/method_comparison/MetaMathQA#readme gradio space: https://huggingface.co/spaces/peft-internal-testing/PEFT-method-comparison
2. Checkout existing related PR for better understand what we are really benchmarking
3. **This PR throwing together multiple distinct things**
4. IMP: I would recommend to mention the use of AI if you did

> Adds a benchmark script that systematically compares 5 PEFT fine-tuning
techniques on conversational AI data, 

Focus on one method to come up with supreme eval results


Wait for **Benjamin** thoughts

Note: Your PR is showing **may close these issues # 2310**, be carefull for this, many works are still going under that, it is a full comparison suite

**Comment by BenjaminBossan** [comment]
@sohammmmm10 Thank you for the PR but as @Adya-Prasad wrote, you misunderstood the type of contribution we expected. There are already two benchmarks but some PEFT methods have experimental settings that are suboptimal or don't have experiments at all; first and foremost, we would like the community to contribute more and better experiments to those benchmarks.

We are open to the idea of adding new benchmarks, but these need to follow very strict criteria. If you want to work on that, please discuss the idea first in #2310. The one you added here is not following these criteria and is not sufficiently different from the existing MetaMath benchmark, so I'll close the PR. Don't feel discouraged to contribute, but please carefully check the type of contribution that is expected.
